hca/dss/__init__.py

```diff
@@ -417,15 +422,31 @@ def _bundle_download_tasks(self,
             if intermediate_path:
                 walking_dir = os.path.join(walking_dir, intermediate_path)
 
-            logger.info("File %s: Retrieving...", filename)
             file_path = os.path.join(walking_dir, filename_base)
-            yield dss_file, functools.partial(self._download_and_link_to_filestore,
+            yield dss_file, functools.partial(self._try_file_checkout,
+                                              executor,
                                               download_dir,
                                               dss_file,
                                               file_path,
                                               num_retries=num_retries,
                                               min_delay_seconds=min_delay_seconds)
 
+    def _try_file_checkout(self, executor, download_dir, dss_file, file_path, num_retries, min_delay_seconds):
+        logger.info('Getting checkout status for file: %s', dss_file.name)
+        response = self.get_file._request(
+            dict(uuid=dss_file.uuid, version=dss_file.version, replica=dss_file.replica),
+            retries=False, stream=True)
+        logger.debug('status_code %s for file %s', response.status_code, dss_file.name)
+        if response.status_code == 301:
+            logger.info('file: %s still checking out...', dss_file.name)
+            executor.submit(self._try_file_checkout, executor, download_dir, dss_file, file_path, num_retries,
+                            min_delay_seconds)
+        elif response.status_code is 302 or 200:
```

Comment:
This should be

```suggestion
        elif response.status_code in (302, 200):
```

The current line is analogous to

```
    elif (response.status_code is 302) or 200:
```

and since `bool(200) == True`, this condition will always be met.